### PR TITLE
[#IOPID-757] logout endpoint

### PR DESCRIPTION
**Pull request by arcogabbo**
This PR depends on https://github.com/pagopa/io-infra/pull/690

#### List of Changes

* logout endpoint

#### Motivation and Context

introduce a new endpoint

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,6 +66,8 @@ The following table contains the required ENV variables that the applicative req
 | LOLLIPOP_GET_ASSERTION_BASE_URL    | API Url for `getAssertion` operation         | string |
 | FAST_LOGIN_AUDIT_CONNECTION_STRING | Audit logs blob connection string            | string |
 | FETCH_TIMEOUT_MS                   | (optional) Fetch Timeout for AbortableFetch  | number |
+| BACKEND_INTERNAL_API_KEY           | API Key to authorize `logout`                | string |
+| BACKEND_INTERNAL_BASE_URL          | API Url for `logout` operation               | string |
```
Comment (BurnedMarshal) [inline README.md:70]: Add these variables if missing in infra of the function

File: src/functions/logout.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      // eslint-disable-next-line sonarjs/no-small-switch
+      switch (response.status) {
+        case 200:
+          return TE.right({ message: "User session deleted" });
+        default:
+          return TE.left(
+            new H.HttpError(
+              `Error while deleting user session: downstream component returned ${response.status}`
+            )
+          );
```
Comment (BurnedMarshal) [inline src/functions/logout.ts]: Can we use a `fromPredicate` instead of a two option case? Nevertheless it will map automatically right and left scenario.

File: src/functions/logout.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      }
+    })
+  );
+
+export const makeLogoutHandler: H.Handler<
+  H.HttpRequest,
+  H.HttpResponse<null, 200>,
```
Comment (BurnedMarshal) [inline src/functions/logout.ts]: What if we change to `204 No Content` ?

File: src/middlewares/__tests__/request.test.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    expect(result).toStrictEqual(aValidPayload);
+  });
+
+  it.each`
+    body
+    ${{}}
+    ${{ fiscalcode: "", sample: "" }}
+    ${{ fiscalcode: aValidPayload.fiscalcode, sample: "" }}
+    ${{ fiscalcode: "", sample: aValidPayload.sample }}
```
Comment (BurnedMarshal) [inline src/middlewares/__tests__/request.test.ts:35]: Add a test with a non empty wrong `fiscalCode`

**Review by BurnedMarshal — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (arcogabbo) [reply README.md:70]: thank you. the relative PR is https://github.com/pagopa/io-infra/pull/690

File: src/functions/logout.ts (hunk 2 above)
Comment (arcogabbo) [reply src/functions/logout.ts]: nice suggestion. applied in https://github.com/pagopa/io-functions-fast-login/pull/17/commits/1c1dc8ff8b69c83ebd5cc6468443c23b2beea5e2

File: src/middlewares/__tests__/request.test.ts (hunk 4 above)
Comment (arcogabbo) [reply src/middlewares/__tests__/request.test.ts:35]: applied in https://github.com/pagopa/io-functions-fast-login/pull/17/commits/6f602199ae9c34bb5776201e57d468f13af8c49f

File: src/functions/logout.ts (hunk 3 above)
Comment (arcogabbo) [reply src/functions/logout.ts]: applied in https://github.com/pagopa/io-functions-fast-login/pull/17/commits/cd3fa6cfb50a4694b6f3d09f58eec1c1ca8567dd

**Review by gquadrati — APPROVED** [review]
LGTM

**Review by BurnedMarshal — APPROVED** [review]
LGTM
